contracts/brush/proc_macros/lib.rs

```diff
@@ -22,41 +22,224 @@ pub fn contract(_attrs: TokenStream, ink_module: TokenStream) -> TokenStream {
+///             self._balances_mut().insert(from, from_balance - amount);
+///             let to_balance = self.balance_of(to);
+///             self._balances_mut().insert(to, to_balance + amount);
+///         }
+///     }
+///
+///     #[ink(storage)]
+///     #[derive(Default, PSP20ExampleStorage)]
+///     pub struct PSP20Struct {
+///         hated_account: AccountId,
+///     }
+///
+///     impl PSP20Example for PSP20Struct {
+///         // Let's override method to reject transactions to bad account
+///         fn _transfer_from_to(&mut self, from: AccountId, to: AccountId, amount: Balance) {
+///             assert!(to != self.hated_account, "I hate this account!");
+///             #[super]self._transfer_from_to(from, to, amount);
+///         }
+///     }
+///
+///     impl PSP20Struct {
+///         #[ink(constructor)]
+///         pub fn new(hated_account: AccountId) -> Self {
+///             let mut instance = Self::default();
+///             instance.hated_account = hated_account;
+///             instance
+///         }
+///     }
+/// }
+/// ```
 #[proc_macro_attribute]
 pub fn trait_definition(_attrs: TokenStream, _input: TokenStream) -> TokenStream {
-    let input: TokenStream2 = _input.clone().into();
-    let attrs: TokenStream2 = _attrs.into();
-    let trait_item = parse_macro_input!(_input as ItemTrait);
-
-    let locked_file = internal::get_locked_file();
-    let mut metadata = internal::load_metadata(&locked_file);
-    internal::put_trait(&mut metadata.external_traits, trait_item);
-    internal::save_metadata_and_unlock(locked_file, metadata);
-
-    let code = quote! {
-        #attrs
-        #[ink_lang::trait_definition]
-        #input
-    };
-    code.into()
+    trait_definition::generate(_attrs, _input)
 }
 
+/// Marks trait definition to brush as special storage trait definition.
+///
+/// This macro stores definition of the trait in a temporary file during build process.
+/// Based on this definition [`#[brush::contract]`](`macro@crate::contract`)
+/// will generate fields and getters for struct, which will derive this trait.
+/// The name of each field is substring between prefix `_` and suffix `_mut`.
+/// The type of the field is the return type of getter(It means that you need to use
+/// the same naming of types in the crate where you will derive this trait).
+///
+/// There are some restrictions that you must follow:
+/// - The trait marked by this macro must contain only the definition of getters
+///   for fields of some structure (no other logic).
+/// - The first character of the name of method must be `_` underscore.
+/// - Each field must contain **exactly** two getters:
+///   - Getter by reference `_field() -> & Type`
+///   - Getter by mut reference `_field_mut() -> &mut Type`
+/// - The getter by mut reference must have the same name as the getter by reference + suffix `_mut`.
+///
+///  ** Note ** This macro must be processed before [`#[brush::contract]`](`macro@crate::contract`),
```

Comment:
The same as above. I think it's more implementation details, than usage. Let's just say that this macro should be inside bruch::contract